tests/nativeapp/test_utils.py

```diff
@@ -0,0 +1,32 @@
+import pytest
+from snowflake.cli.api.secure_path import SecurePath
+from snowflake.cli.plugins.nativeapp.utils import get_first_paragraph_from_markdown_file
+
+
+@pytest.mark.parametrize(
+    "file_content, expected_paragraph",
+    [
+        (
+            """
+            ## Introduction
+
+            This is an example template for a Snowflake Native App project which demonstrates the use of Python extension code and adding Streamlit code. This template is meant to guide developers towards a possible project structure on the basis of functionality, as well as to indicate the contents of some common and useful files. 
+
+            Since this template contains Python files only, you do not need to perform any additional steps to build the source code. You can directly go to the next section. However, if there were any source code that needed to be built, you must manually perform the build steps here before proceeding to the next section. 
+
+            Similarly, you can also use your own build steps for any other languages supported by Snowflake that you wish to write your code in. For more information on supported languages, visit [docs](https://docs.snowflake.com/en/developer-guide/stored-procedures-vs-udfs#label-sp-udf-languages).
+            """,
+            "This is an example template for a Snowflake Native App project which demonstrates the use of Python extension code and adding Streamlit code. This template is meant to guide developers towards a possible project structure on the basis of functionality, as well as to indicate the contents of some common and useful files.",
+        )
+    ],
+)
+def test_get_first_paragraph_from_markdown_file(file_content, expected_paragraph):
+    with SecurePath.temporary_directory() as temp_path:
+        temp_readme_path = (temp_path / "README.md").path
+
+        with open(temp_readme_path, "w+") as temp_readme_file:
+            temp_readme_file.write(file_content)
+
+        actual_paragraph = get_first_paragraph_from_markdown_file(temp_readme_path)
```

Comment:
The bulk of this test should be within the `with` statement; the file can be deleted at any time after you exit its context manager

Reply:
I actually had the opposite problem where if I put the two lines inside the context manager, then the tests fail because the function still reads the file as if it was empty:

![image](https://github.com/snowflakedb/snowflake-cli/assets/156030333/a76aea29-ed49-4d63-b4ce-c6a7424109b2)

I'm guessing Python doesn't flush the file changes until the `open` context manager is exited?

I would've thought that only after exiting the top-level context manager with the temporary directory that the temp files would be deleted 🤔

Reply:
You're right.

I misread the command, we had one that would scope the file's lifetime to the same `with` statement as opening it. But that code has since been refactored.